### PR TITLE
CGT-1675: Refactored common.Dates

### PR DESCRIPTION
**Pull request by MJCallahanPage**

### CODE REVIEW

File: app/common/Dates.scala [hunk 1]
```diff
@@ -28,69 +28,47 @@ object Dates {
...
-    date match {
-      case Some(date) =>
-        val cal = Calendar.getInstance()
-        cal.setTime(date)
-        cal.add(Calendar.MONTH, months * -1)
-        new SimpleDateFormat("d MMMM yyyy").format(cal.getTime)
-      case _ => ""
+  def dateMinusMonths(date: Option[Date], months: Int): String =
+    date.fold(""){a =>
+      val cal = Calendar.getInstance()
```
Comment (DavidWoodward37) [inline app/common/Dates.scala]: You could remove this val cal too now you made an instance for the object :)

**Comment by DavidWoodward37** [comment]
This was bad task writing on my part but could you refactor to use jodatime instead of java.util.Date? 

Apologies.

**Comment by MJCallahanPage** [comment]
Refactored to use `java.time` instead of `java.util.date`

I investigated joda.time but on their site they recommend for uses of java 8 onward to us java.time as this is a core part of the JDK.

This change required a large refactor

**Comment by jameshforster** [comment]
Could you fix the merge conflicts please.

**Comment by MJCallahanPage** [comment]
@jameshforster : resolved conflicts

**Pull request merged**